bot/cogs/doc/cog.py

```diff
@@ -189,6 +260,19 @@ async def refresh_inventory(self) -> None:
         ]
         await asyncio.gather(*coros)
 
+    async def get_symbol_markdown(self, symbol_data: DocItem):
+        """TODO"""
+        if symbol_data.name not in self.objects:
+            q = self.urls[symbol_data.base_url + symbol_data.relative_url_path]
+            if not q.started:
+                async with self.bot.http_session.get(symbol_data.url) as response:
+                    soup = BeautifulSoup(await response.text(encoding="utf8"), 'lxml')
+                    soup.find("head").decompose()
```

Comment:
I find this conditional behaviour to be a confusing API for the class i.e. the fact that `soup` is only needed sometimes. I think it'll be clearer if this is done inside the class to simplify the API.